parse_date.re

```diff
@@ -1353,6 +1355,18 @@ weekdayof        = (reltextnumber|reltexttext) space (dayfull|dayabbr) space 'of
 		return TIMELIB_ISO_DATE;
 	}
 
+	iso8601datex
+	{
+		DEBUG_OUTPUT("iso8601datex");
+		TIMELIB_INIT;
+		TIMELIB_HAVE_DATE();
+		s->time->y = timelib_get_unsigned_nr((char **) &ptr, 20);
```

Comment:
You're only reading up to 20 characters, so the regular expression in line 911 should also be up to 20 only.